classes/Utils/Upgrades.php

```diff
@@ -157,7 +157,7 @@ public static function update_plugin_version() {
 
 			// Reset JS/CSS assets for regeneration.
 			pum_reset_assets();
-		} elseif ( ! self::$upgraded_from || self::$upgraded_from === 'false' ) {
+		} elseif ( ! self::$upgraded_from || self::$upgraded_from === 'false' ) {  // phpcs:ignore
```

Comment:
@s-acv2 - Not sure the confusion here, but guessing because it appears to be checking for false already you assume something janky going on here. You'd be partially correct, we are also checking in case `false` was passed as a `string` instead of a `boolean`.

```
! self::$upgraded_from || 'false' === self::$upgraded_from
```